service/request.go

```diff
@@ -0,0 +1,199 @@
+package service
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/lake/journal"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zio/anyio"
+	"github.com/brimdata/zed/zqe"
+	"github.com/brimdata/zed/zson"
+	"github.com/gorilla/mux"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/zap"
+)
+
+type Request struct {
+	*http.Request
+	Logger *zap.Logger
+}
+
+func newRequest(w http.ResponseWriter, r *http.Request, logger *zap.Logger) (*ResponseWriter, *Request) {
+	logger = logger.With(zap.String("request_id", api.RequestIDFromContext(r.Context())))
+	req := &Request{
+		Request: r,
+		Logger:  logger,
+	}
+	m := zson.NewZNGMarshaler()
+	m.Decorate(zson.StyleSimple)
+	res := &ResponseWriter{
+		ResponseWriter: w,
+		Logger:         logger,
+		marshaler:      m,
+	}
+	accept := r.Header.Get("Accept")
+	if accept == "" {
+		accept = "application/json"
+	}
+	res.SetContentType(accept)
+	return res, req
+}
+
+func (r *Request) PoolID(w *ResponseWriter) (ksuid.KSUID, bool) {
+	return r.TagFromPath("pool", w)
+}
+
+func (r *Request) TagFromPath(arg string, w *ResponseWriter) (ksuid.KSUID, bool) {
```

Comment:
`gorilla/mux` refers to these things as route variables or URL variables. (I think route is better because it's more specific than URL.)
```suggestion
func (r *Request) RouteVar(arg string, w *ResponseWriter) (ksuid.KSUID, bool) {
```